src/node/stage/approved.rs

```diff
@@ -1681,6 +1681,12 @@ impl Approved {
         self.chain.is_self_elder() || req.elders_version < self.chain.our_info().version()
     }
 
+    // If elder, always handle UserMessage, otherwise handle it only if addressed directly to us
+    // as a node.
+    fn should_handle_user_message(&self, our_id: &PublicId, dst: &DstLocation) -> bool {
+        self.chain.is_self_elder() || dst.as_node().ok() == Some(our_id.name())
```

Comment:
why not sue `Ok(our_id.name())` so that the convertion of `ok()` (Result -> Option) can be skipped?

Reply:
Good point, will change.

Reply:
Actually no, it doesn't work because `Result<_, RoutingError>` doesn't implement `Eq` (probably because `RoutingError` doesn't implement it.

Reply:
Ah, Ok then.